src/Couchbase.Lite.Tests.Shared/DocumentTest.cs

```diff
@@ -2100,6 +2100,70 @@ public void TestRevisionIDExistingDoc()
             }
         }
 
+        [Fact]
+        public void TestTypesInDocumentToJSON()
+        {
+            var dic = PopulateDictData();
+            using (var md = new MutableDocument("doc1")) {
+                foreach (var item in dic) {
+                    switch (item.Value) {
+                        case null:
+                            md.SetValue(item.Key, item.Value);
+                            break;
+                        case int i:
+                            md.SetInt(item.Key, i);
+                            break;
+                        case string str:
+                            md.SetString(item.Key, str);
+                            break;
+                        case long l:
+                            md.SetLong(item.Key, l);
+                            break;
+                        case bool bl:
+                            md.SetBoolean(item.Key, bl);
+                            break;
+                        case float f:
+                            md.SetFloat(item.Key, f);
+                            break;
+                        case double d:
+                            md.SetDouble(item.Key, d);
+                            break;
+                        case Blob blob:
+                            md.SetBlob(item.Key, blob);
+                            break;
+                        case DateTimeOffset dto:
+                            md.SetDate(item.Key, dto);
+                            break;
+                        case DictionaryObject cbldo:
+                            md.SetDictionary(item.Key, cbldo);
+                            break;
+                        case ArrayObject cblarr:
+                            md.SetArray(item.Key, cblarr);
+                            break;
+                        case IDictionary<string, object> dict:
+                        case int[] ao:
+                        case ulong ul:
+                        case byte b:
+                        case sbyte sb:
+                        case ushort us:
+                        case short s:
+                        case uint ui:
+                            md.SetValue(item.Key, item.Value);
+                            break;
+                    }
+                }
+
+                Db.Save(md);
+            }
+
+            using (var doc = Db.GetDocument("doc1")) {
+                var json = doc.ToJSON();
+                var jdic = JsonConvert.DeserializeObject<DataInCBLDataType>(json);
+
+                VerifyValuesInJson(dic, jdic);
```

Comment:
Isn't this testing that a blob is inside?  The ToJSON call should be returning a dictionary in place of a Blob but is that being tested?

Reply:
The test is actually to test Document ToJSON with all the CBL support types, and Blob is one of it.

Originally I have 2 branches, one blob ToJSON related (PR title reflects that) and the other one is Document and Dictionary ToJSON and it's branched from blob ToJSON. When I push the later branch, somehow it merged with the Blob PR...

Reply:
From what I understood, the document JSON in ToJSON cannot have an actual blob inside of it, it needs to be all standard JSON types, which means the blob will be a dictionary.  That might be happening implicitly here but I can't tell.

Reply:
Updated the VerifyValuesInJson to specifically run Blobs ToJSON and deserialize and compare both json values.